x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_migration_service.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Crypto } from '@elastic/node-crypto';
```

Comment:
So there's a couple of things here:
1. Regarding naming - that's fair, I was looking for a name that would clarify that it's uses different version of types and to make sure it isn't confused with the regular service somehow - how would you name it? If we decide to merge it with the service, this won't matter, but if we don't merge them, I'm happy to rename this as whatever you'd like. :)
2. Regarding the separation, I actually had the same concern and mentioned it to @legrego while working on this. It's up to you, as you'll be the ones maintaining this code, but it felt like that would be harder to maintain than the duplication, so I'll let you decide that.
3. Regarding registering legacy types, I'd have to look into how that works under the hood, but my thinking was that by moving this type from the registration into the migration function was that it would mean that the old type, which is only needed for migration purposes, is then coupled with the migration code in the plugin which leaves the plugins that use ESO cleaner. Wouldn't registering old versions make the registration more complicated? As you'd have to specify version numbers etc. 🤔